src/main/java/resources/PairResource.java

```diff
@@ -30,14 +31,16 @@ public PairResource(SlotDao slotDao) {
     @UnitOfWork
     public List<Person> findPairs(@QueryParam("startDate") String startDate,
                                   @QueryParam("endDate") String endDate,
+                                  @QueryParam("startTime") Time startTime,
+                                  @QueryParam("endTime") Time endTime,
                                   @QueryParam("isDev") Boolean isDev,
                                   @QueryParam("isTest") Boolean isTest,
                                   @QueryParam("isOps") Boolean isOps,
                                   @QueryParam("isOther") Boolean isOther) {
 
-        List<Slot> slots = slotDao.findSlotsForPairMatching(startDate, endDate, isDev, isTest, isOps, isOther);
-        PairFinder pf = new PairFinder();
+        List<Slot> slots = slotDao.findSlotsForPairMatching(startDate, endDate, startTime, endTime, isDev, isTest, isOps, isOther);
+        PairFinder pairFinder = new PairFinder();
```

Comment:
`PairFinder` can be an instance field of the resource. It would be also good to inject it via constructor.